mlir/lib/Target/LLVM/NVVM/Target.cpp

```diff
@@ -572,6 +572,9 @@ NVPTXSerializer::moduleToObject(llvm::Module &llvmModule) {
     getOperation().emitError() << "Failed translating the module to ISA.";
     return std::nullopt;
   }
+  if (isaCallback) {
```

Comment:
We have also `dump-sass`. Should we also add a callback for that?

Reply:
Could we dump the sass result ourselves after getting it?

I think the reason for adding callback functions here is that we cannot obtain the intermediate results (LLVM IR, etc.) by subclass `SerializeGPUModuleBase` (maybe we can override `ModuleToObject::run()`, but this will cause code duplication, which is not a good solution).

Reply:
Sure we can 😀 But this means code replication. 

This path calls nvdisasm on cubin and print sass. I think it's better to leverage this mechanism, and all the downstream compilers use it.

Reply:
I don't quite get it? The callback would return the same thing as the serializer?

Reply:
> This means code replication. This path calls nvdisasm on cubin and print sass. I think it's better to leverage this mechanism, and all the downstream compilers use it.

I think you are right. But in my current understanding, `sass` is a concept that only exists in `nvvm`. So we can not name a callback `sassCallBack`. Similarly, we cannot name a callback `ptxCallback`, but we can name it `isaCallback`.